cli/internal/globby/globby.go

```diff
@@ -1,55 +1,120 @@
 package globby
 
 import (
-	"os"
+	"errors"
 	"path/filepath"
 	"strings"
 
+	iofs "io/fs"
+
 	"github.com/bmatcuk/doublestar/v4"
 	"github.com/spf13/afero"
+	"github.com/vercel/turborepo/cli/internal/util"
 )
 
-var _osFS = afero.NewOsFs()
+var _aferoOsFs = afero.NewOsFs()
+var _aferoIOFS = afero.NewIOFS(_aferoOsFs)
 
 func GlobFiles(basePath string, includePatterns []string, excludePatterns []string) []string {
-	return globFilesFs(_osFS, basePath, includePatterns, excludePatterns)
+	return globFilesFs(_aferoIOFS, basePath, includePatterns, excludePatterns)
+}
+
+// isRelativePath ensures that the the requested file path is a child of `from`.
+func isRelativePath(from string, to string) (isRelative bool, err error) {
```

Comment:
This is somewhat a judgment call and I didn't find anything explicitly in the style guide, so push back if you disagree (also not really a blocker):

The community has a range of opinions on named return values, and their sometimes-companion, the naked `return`[0][1]. I think it's unfortunate that the documentation aspects incur the downsides that enable naked `return`. As a result, my preference is to avoid named return values altogether, as I think they tend to end up less readable and subject to unintentional shadowing. 

I did find a linter, [nakedret](https://github.com/alexkohler/nakedret) that catches half of this problem, and I would be happy to enable if you're on board.

In this particular case, I can see the intention to label a `bool` return value. My opinion is that the function name and associated comment is sufficient, but therein lies the judgment call. I also think you have shadowed `err` via `:=` assignment, which works out in this case, but may not have been your intention(?). 

Sample of opinions:
0 : https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values
1 : https://github.com/golang/go/issues/21291

Reply:
I ... did not fully understand how that worked when I wrote it. 😅 I was treating it as a "typed tuple" for the return values with "naming hints" without needing to go so far as actually creating a struct.

<img width="562" alt="Name Hints" src="https://user-images.githubusercontent.com/20542/165662074-b0443645-2250-4c50-bc21-18dc3ecb5029.png">

I believe naked returns in a function with any branching is not a particularly great idea and prefer explicit over implicit. I'd *love* to enable `nakedret` with `l` set to something silly like `3`. ("If you can fit a naked return in a function this size, you know what, have at it.")

I also found `shadow` which might help me not shoot my foot off:

```
go install golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow@latest
go vet -vettool=$(which shadow) globby.go # 0 right now
```

Thoughts on somehow enabling that as a linter?

***

I'm going to move this away from named return values, I agree that it's really easy to accidentally stumble into a minefield with this approach.